### PR TITLE
Tracking full chain

### PR DESCRIPTION
**Pull request by pbutti**
love you omar

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,13 +48,47 @@ if(BUILD_EVENT_ONLY)
      return()
 endif()
 
+# since ACTS changes frequently enough, we build it as a submodule of Tracking
+#   here we 'set' some CMake variables that we would have passed to 'cmake' on
+#   the command line with '-D' if we were building ACTS directly.
+set(ACTS_BUILD_PLUGIN_DD4HEP ON) # representing geometry with DD4hep right now
+set(ACTS_BUILD_PLUGIN_TGEO ON) # representing geometry with DD4hep right now
```
Comment (tomeichlersmith) [inline CMakeLists.txt:55]: We don't use DD4hep anymore :tada:

```suggestion
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+Installation of ldmx-sw from docker with acts
```
Comment (tomeichlersmith) [inline README.md:1]: Will want to update this...

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "acts"]
+	path = acts
+	url = https://github.com/acts-project/acts
```
Comment (tomeichlersmith) [inline .gitmodules:3]: Is acts on a tag? Just want it to be very obvious what version of acts is being used

File: CMakeLists.txt [hunk 4]
```diff
@@ -38,13 +48,47 @@ if(BUILD_EVENT_ONLY)
...
+set(ACTS_BUILD_EXAMPLES OFF)     # don't waste time building examples
+set(CMAKE_CXX_STANDARD 17)       # set C++ standard we use within ldmx-sw
+add_subdirectory(acts)           # now build acts
+#add_subdirectory(acts-dd4hep)
+# luckily for us, adding ACTS as a subdirectory produces the same CMake targets
+#   as find_package, so we don't need to do anything else from here on out
+
+find_package(Eigen3 CONFIG REQUIRED)
+
+setup_dd4hep_target()
```
Comment (tomeichlersmith) [inline CMakeLists.txt:65]:
```suggestion
```

**Review by tomeichlersmith — COMMENTED** [review]
**First pass review** just looking at surface stuff and not inspecting the actual tracking.

## General To Do
- [ ] Expunge all references to DD4hep
- [ ] Make sure you use spaces instead of tabs. I know you're an emacs-dood so I don't know how to help you. We use tab = 2 spaces.
- [ ] Format the C++ files. `find . -type f -name '*.h' -o -name '*.cxx' -exec clang-format -i --style=Google {} ';'`

**Comment by pbutti** [comment]
In order to keep track of the validation done, I attach the links to the validation rootFiles produced by the reconstruction using the two geometries, based on DD4hep and GDML, respectively.
DD4Hep: https://www.dropbox.com/sh/zfgh3sfp0p37hqw/AAAfDqI9rreZiaOh-DKBq5fma?dl=0
GDML (G4Parser) : https://www.dropbox.com/sh/wr7htwyh3dgnd5g/AABpu2xOsSwucpUeJ6bQZatma?dl=0

I checked the track parameter residuals and they are all consistent. Minor changes can be due to the randomization in the smearing and small changes in the positioning of the sensors (now correct in the G4Parser version)

File: CMakeLists.txt (hunk 1 above)
Comment (pbutti) [reply CMakeLists.txt]: Done.

File: README.md (hunk 2 above)
Comment (pbutti) [reply README.md]: Done

File: .gitmodules (hunk 3 above)
Comment (pbutti) [reply .gitmodules:3]: I'm planning to move from tag 19.6 to 20. But that will require work and validation.

File: CMakeLists.txt (hunk 4 above)
Comment (pbutti) [reply CMakeLists.txt]: Removed

File: .gitmodules (hunk 3 above)
Comment (pbutti) [reply .gitmodules:3]: Could you change this to pick version v19.6.0

**Review by omar-moreno — APPROVED** [review]
There are a few things that need to be cleaned up and docs need to be added but these can be done via a patch.  After fixing the python configs, this PR can be merged.

**Comment by omar-moreno** [comment]
After looking at this PR further, there are quite a few things that need to be fixed. Given the size of this PR and the fact that it's working, I'm going to merge it into trunk and open a separate issue which outlines the updates that are needed.

**Pull request merged**